frontend/ui/otamanager.lua

```diff
@@ -456,7 +456,7 @@ function OTAManager:getOTAMenuTable()
         text = _("Update"),
         hold_callback = function()
             if not NetworkMgr:isOnline() then
-                NetworkMgr:promptWifiOn()
+                NetworkMgr:beforeWifiAction(OTAManager.fetchAndProcessUpdate)
```

Comment:
You might want to provide as the callback an anonymous `function() OTAManager:fetchAndProcessUpdate() end`.
You're lossing the reference to sefl with just `OTAManager.fetchAndProcessUpdate` - which may be is fine with this particular one, but won't be with most real methods of a real object :)